whiteout/coalescer.go

```diff
@@ -0,0 +1,28 @@
+package whiteout
+
+import (
+	"context"
+
+	"github.com/quay/claircore"
+	"github.com/quay/claircore/indexer"
+)
+
+type Coalescer struct{}
+
+// NewCoalescer is a constructor for a Coalescer
+func NewCoalescer() *Coalescer {
+	return &Coalescer{}
```

Comment:
For Java, I think this was switched to start using `nil` instead of an empty struct. Is that a pattern we want to continue? @hdonnay

Reply:
Yeah, if it's effectively the zero struct, using nil is basically the same as using a pointer to a (the) literal

Reply:
:+1: